app/javascript/js/controllers.js

```diff
@@ -53,59 +53,78 @@ import ToggleController from './controllers/toggle_controller'
-application.register('dashboard-card', DashboardCardController)
-application.register('date-time-filter', DateTimeFilterController)
-application.register('filter', FilterController)
-application.register('form', FormController)
-application.register('hidden-input', HiddenInputController)
-application.register('input-autofocus', InputAutofocusController)
-application.register('item-select-all', ItemSelectAllController)
-application.register('item-selector', ItemSelectorController)
-application.register('loading-button', LoadingButtonController)
-application.register('media-library-attach', MediaLibraryAttachController)
-application.register('media-library', MediaLibraryController)
-application.register('menu', MenuController)
-application.register('modal', ModalController)
-application.register('multiple-select-filter', MultipleSelectFilterController)
-application.register('avo-nested-form', NestedFormController)
-application.register('panel-refresh', PanelRefreshController)
-application.register('per-page', PerPageController)
-application.register('preview', PreviewController)
-application.register('record-selector', RecordSelectorController)
-application.register('resource-edit', ResourceEditController)
-application.register('resource-index', ResourceIndexController)
-application.register('resource-show', ResourceShowController)
-application.register('search', SearchController)
-application.register('select-filter', SelectFilterController)
-application.register('select', SelectController)
-application.register('self-destroy', SelfDestroyController)
-application.register('sidebar', SidebarController)
-application.register('sign-out', SignOutController)
-application.register('table-row', TableRowController)
-application.register('tabs', TabsController)
-application.register('text-filter', TextFilterController)
-application.register('tippy', TippyController)
-application.register('toggle', ToggleController)
-application.register('trix-body', TrixBodyController)
+// Special cases for controller registration names
+const SPECIAL_CASES = {
+  nested_form: 'avo-nested-form'
+}
 
-// Field controllers
-application.register('belongs-to-field', BelongsToFieldController)
-application.register('code-field', CodeFieldController)
-application.register('date-field', DateFieldController)
-application.register('easy-mde', EasyMdeController)
-application.register('key-value', KeyValueController)
-application.register('progress-bar-field', ProgressBarFieldController)
-application.register('reload-belongs-to-field', ReloadBelongsToFieldController)
-application.register('tags-field', TagsFieldController)
-application.register('tiptap-field', TiptapFieldController)
-application.register('trix-field', TrixFieldController)
+// Helper function to convert snake_case to kebab-case
+function toKebabCase(str) {
+  return str.replace(/_/g, '-')
+}
+
+// Controllers registry - automatically registers all controllers
+const CONTROLLERS = [
+  ['action', ActionController],
```

Comment:
And we then can infer the names.
So in teh end we'd end up with somethign like this

```js
import ActionController from ...
import ...
import ...

registerControllers(application, [Actioncontroller, ..., ...])


it would be easier for us to implement this as well
https://github.com/avo-hq/avo/issues/2030